frontend/src/pages/notebookController/NotebookControlPanelRedirect.tsx

```diff
@@ -0,0 +1,32 @@
+import * as React from 'react';
+import { useParams, useHistory } from 'react-router-dom';
+import ApplicationsPage from 'pages/ApplicationsPage';
+import { getUserStateFromDashboardConfig } from 'utilities/notebookControllerUtils';
+import AppContext from '../../app/AppContext';
+import NotebookControllerContext from './NotebookControllerContext';
+
+const NotebookControlPanelRedirect: React.FC = () => {
+  const history = useHistory();
+  const { username: translatedUsername } = useParams<{ username: string }>();
+  const { dashboardConfig } = React.useContext(AppContext);
+  const { setCurrentUserState } = React.useContext(NotebookControllerContext);
+  React.useEffect(() => {
+    if (translatedUsername && dashboardConfig.spec.notebookController?.enabled) {
+      const userState = getUserStateFromDashboardConfig(
+        translatedUsername,
+        dashboardConfig.status?.notebookControllerState,
+      );
+      if (!userState) {
+        history.push('/not-found');
+      } else {
+        setCurrentUserState(userState);
+        history.push('/notebookController');
```

Comment:
I'm thinking for successful states, we'll want to just replace the URL as to not complicate the browser history.
```suggestion
        history.replace('/notebookController');
```

Reply:
Updated.